### ISSUE TITLE
lock exceptions: display more infos

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#2261]
can we display more information about why a lock could not be acquired, like e.g. hostname and pid of the lock holder?

**Comment by knutov** [comment]
Just as an idea: May be it's possible (I think this should be not hard to implement) to write to the lock file (or to the neighbor file) who made lock and why ? So the script will be able to print enough info to understand what to do.

**Comment by ThomasWaldmann** [comment]
Yes, theoretically possible. But while we already have hostname/pid in the lockfile, we do not have "reason" yet.

Also, using ps you could find out more details yourself until we have that - if you can't already do a good guess via the given hostname.

**Comment by knutov** [comment]
Yes, pid is technically enough, if user is "linux administrator" and familiar with `ps`. Even in this case print "was mounted" can save some seconds on typing ps and etc.

**Comment by enkore** [comment]
PID, hostname are there, could be shown.

Borg 1.1.x (beta) has code that eliminates stale locks in most cases w/ no user intervention.

**Comment by scolby33** [comment]
I think this is a good idea. I came up with it independently just now and found this issue.

An example use case: the borg repo is being backed up to external storage with `rsync`, `rclone`, etc. Something like
```
server$ borg with-lock --message "rsync backup in progress" $repo rsync ...
```
```
client$ borg create ...
Failed to create/acquire the lock ...(timeout)
"rsync backup in progress"
```
could be very useful, I think.

Right now it looks like the lock files are 0 bytes created like `touch` or similar. Perhaps the `--message` option could add content to the file which is then printed whenever a lock is found (and maybe even with a prompt before allowing it to be broken?)
